app/models/location.rb

```diff
@@ -134,6 +148,27 @@ class Location < AbstractModel
     end
   end
 
+  # TODO: write tests for :in_region, :in_box, and any other untested scopes.
+  # Coveralls has info. E.g., https://coveralls.io/builds/51093827
+
+  scope :name_includes,
+        ->(place_name) { where(Location[:name].matches("%#{place_name}%")) }
+  scope :in_region,
+        ->(place_name) { where(Location[:name].matches("%#{place_name}")) }
+  scope :in_box, # Use named parameters (n, s, e, w), any order
+        lambda { |**args|
+          if args[:s].present? && args[:n].present? &&
+             args[:w].present? && args[:e].present? &&
+             (args[:w] < args[:e])
+
+            where(
+              (Location[:south] >= args[:s]).and(Location[:north] <= args[:n]).
+              and(Location[:west] >= args[:w]).and(Location[:east] <= args[:e]).
+              and(Location[:west] <= Location[:east])
+            )
+          end
+        }
+
```

Comment:
@pellaea: Two Issues re `Location.in_box`
1. Is this literal edge case is a Ruby &/or SQL floating point thing?
Should I try to deal with it by automatically tweaking the arguments by a small value?
```ruby
(ruby) cal = locations(:california)
(ruby) locs_in_cal_box.include?(cal)
false
(ruby) locs_in_cal_box.pluck(:name)
["Howarth Park, Santa Rosa, California, USA",
 "\"Mitrula Marsh\", Sand Lake, Bassetts, Yuba Co., California, USA",
 "Point Reyes National Seashore, Marin Co., California, USA",
 "Burbank, California, USA",
 "Salt Point State Park, Sonoma Co., California, USA",
 "Gualala, California, USA",
 "Albion, California, USA"]
(ruby) locs_in_cal_box_plus = Location.in_box(n: cal.north + 0.00001, s: cal.south - 0.00001, e: cal.east + 0.00001, w: cal.west - 0.00001)
(ruby) locs_in_cal_box_plus.include?(cal)
true
```
2. This scope will return `nil` for boxes straddling 180 degrees. 
Should I fix that. (I'm inclined to fix it.)

Reply:
1. Yes, that sounds like a rounding issue, and the epsilon tweak is probably the standard solution, good call.
2. Yes, please do fix it to work for boxes straddling the dateline (i.e. west > east).  I'm surprised I didn't catch that.  Every box comparison always has to cover both cases: one for west < east, one for west > east.